testscript/cmd.go

```diff
@@ -492,7 +493,50 @@ func (ts *TestScript) cmdUNIX2DOS(neg bool, args []string) {
 	}
 }
 
-// Tait waits for background commands to exit, setting stderr and stdout to their result.
+// cmdKill kills background commands.
+func (ts *TestScript) cmdKill(neg bool, args []string) {
+	if len(args) > 1 {
+		ts.Fatalf("usage: kill [name]")
+	}
+	if neg {
+		ts.Fatalf("unsupported: ! kill")
+	}
+	if len(args) > 0 {
+		ts.killBackgroundOne(args[0])
+	} else {
+		ts.killBackground()
+	}
+}
+
+func (ts *TestScript) killBackgroundOne(bgName string) {
+	bg := ts.findBackground(bgName)
+	if bg == nil {
+		ts.Fatalf("unknown background process %q", bgName)
+	}
+	signal := os.Interrupt
```

Comment:
Might be more generic to be able to choose the signal? (hup vs int vs kill...)

Reply:
Yeah, I was thinking that. I was wondering about the API for that. Would you be happy with `kill -SIG [name]`? (SIG would be numeric only in the first instance).